scripts/create-dsa-account.js

```diff
@@ -1,20 +1,59 @@
-const Web3 = require('web3');
 const DSA = require('dsa-connect');
+require("@nomiclabs/hardhat-web3");
+require("hardhat")
 require('dotenv').config();
-const ETH_NODE_URL = `https://eth-mainnet.alchemyapi.io/v2/${process.env.ALCHEMY_KEY}`;
-const web3 = new Web3(new Web3.providers.HttpProvider(ETH_NODE_URL)); // to enable web3 calls via SDK
-
-const dsa = new DSA({
-    web3: web3,
-    mode: "node",
-    privateKey: process.env.PRIVATE_KEY
-});
-
-const buildAccount = async () => {
-    const account = await dsa.build({
-        gasPrice: web3.utils.toWei('0', 'ether') // estimated gas price
-        // origin: '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266',
-        // authority: '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266'
+
+const address = '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266';
+
+const setBalance = async () => {
+    await network.provider.send("hardhat_setBalance", [
+        address,
+        "0xD3C21BCECCEDA1000000"
+    ]);
+}
+
+setBalance();
+
+const castSpells = async () => {
+    const dsa = new DSA({
+        web3: web3,
+        mode: "node",
+        privateKey: process.env.PRIVATE_KEY
+    });
+    await dsa.build({
+        gasPrice: web3.utils.toWei('0.000000100000000000', 'ether'),// estimated gas price
+        origin: address,
+        authority: address
     });
+
+    const accounts = await dsa.getAccounts(address);
+    // console.log(accounts[0].id);
+    const dsaId = accounts[0].id;
+    await dsa.setInstance(dsaId);
+    let spells = dsa.Spell({
+        gasPrice: web3.utils.toWei('0.000000100000000000', 'ether'),// estimated gas price
+    });
+
```

Comment:
Your dsa account should have ETH to deposit. The `hardhat_setbalance` you did about was for your metamask address to pay for gas. 
But you have to do that again for your dsa address.
```suggestion
  network.provider.send("hardhat_setBalance", [
    accounts[0]["address"],
    ethers.utils.parseEther("10.0").toHexString(),
  ]);

```